### DISCUSSION TITLE
Integration of TSN with 5G

### DISCUSSION THREAD

**Discussion by Adii294** [omnetpp/omnetpp#997] [Q&A]
Hello Everyone,
This is me ADIL MUHAMMAD, I have trying to connect TSN device with UE but its giving me **findModuleByPath()** error, I am using Omnet 6.0.1, INET 4.5.0 and Simu5G 1.2.2. Additionally, if there is someone working on similar research area then your collaboration would be highly welcome.

**Comment by rhornig** [comment]
There is absolutely no way that someone could help you without actually providing at least the error message and the code you are trying to run.

**Comment by Adii294** [comment]
Sorry for incomplete message here is the ned file of a very simple network in which I am trying to connect the TSN device with UE.
Ned file. 
```
package networkfivegtwo;
import inet.node.tsn.TsnDevice;
import simu5g.nodes.Ue;
import inet.node.ethernet.Eth10G;
import inet.node.inet.StandardHost;

network tsnfiveg
{
    types:
        module UE extends Ue
        {
             gates:
       				 inout ethg @labels(EtherFrame-conn) @allowUnconnected @packable;
        }
  
    submodules:
        tsn1: TsnDevice {
            @display("p=800,400");
            
        }
    	ue1: UE {
    		@display("p=1000,400");
    	}
    connections allowunconnected:
        
              tsn1.ethg++ <--> ue1.ethg;
        
}
```
Here is the error:
```
**cSimulation::getModuleByPath(): Module at 'binder' not found (Note: operation of getModuleByPath() has changed in OMNeT++ version 6.0, use findModuleByPath() in you want the original behavior) -- in module (IP2Nic) tsnfiveg.ue1.cellularNic.ip2nic (id=92), during network initialization**
```
I get prompted for number UE devices and then channel bit rate, afterward I receive the above mentioned error


I am very new to OMNET, I dont know where this error belongs, I tried a lot.

**Reply by rhornig** [reply]
Last month we spent a bit of time on this topic (TSN and 5G integration). We were able to create a simple example, but it is indeed not as straightforward as we would love to see. The short story:

Simu5G uses ppp links to communicate between base stations (i.e. no ethernet interfaces). On the other hand, TSN is obviously working only with ethernet links. We had to add an ethernet interface card to the Simu5G eNode-s. After that we were able to connect some TSN switches directly to the base stations.

**Reply by dddcurry** [reply]
I have also been studying the integration of TSN and 5G networks recently. Could you share your OMNeT++ project so that we can research it together? Thank you.

**Reply by avarga** [reply]
@dddcurry I am working in the [inet-framework/simu5g](https://github.com/inet-framework/simu5g) repo to rectify some architectural issues with Simu5G, and then move on to add TSN-related functionality.
